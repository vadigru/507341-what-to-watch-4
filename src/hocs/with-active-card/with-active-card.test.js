import React from "react";
import withActiveCard from "./with-active-card.jsx";
import renderer from "react-test-renderer";

const FilmData = {
  TITLE: `Joker`,
  GENRE: `Drama`,
  YEAR: 2019
};

const movies = [
  {
    title: `Movie title`,
    posterUrl: `https://url.com/poster.jpg`,
    backgroundUrl: `https://url.com/poster/1.jpg`,
    previewUrl: `https://url.com/preview/video.mp4`,
    genre: `Movie Genre`,
    release: 2020,
    director: `Director Name`,
    starring: [`Actor 1`, `Actor 2`, `Actor 3`],
    time: `1h 00m`,
    rating: 10,
    votes: 1000,
    description: `Movie description`,
    reviews: [
      {
        date: `June 25, 2020`,
        user: `John Doe`,
        comment: `Comment text.`,
        rating: 8.9
      },
    ]
  }
];

const movie = {
  title: `Movie title`,
  posterUrl: `https://url.com/poster.jpg`,
  backgroundUrl: `https://url.com/poster/1.jpg`,
  previewUrl: `https://url.com/preview/video.mp4`,
  genre: `Movie Genre`,
  release: 2020,
  director: `Director Name`,
  starring: [`Actor 1`, `Actor 2`, `Actor 3`],
  time: `1h 00m`,
  rating: 10,
  votes: 1000,
  description: `Movie description`,
  reviews: [
    {
      date: `June 25, 2020`,
      user: `John Doe`,
      comment: `Comment text.`,
      rating: 8.9
    },
  ]
};

const mockComponent = () => <div />;

const MockComponentWrapped = withActiveCard(mockComponent);

it(`Should render MoviePage component`, ()=>{
  const tree = renderer.create(
      <MockComponentWrapped
        movies={movies}
        movie={movie}
        onMovieCardClick={()=>{}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Should render Main component`, ()=>{
  const tree = renderer.create(
      <MockComponentWrapped
        promoTitle = {FilmData.TITLE}
        promoGenre = {FilmData.GENRE}
        promoYear = {FilmData.YEAR}
        movies = {movies}
        onMovieCardClick={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
