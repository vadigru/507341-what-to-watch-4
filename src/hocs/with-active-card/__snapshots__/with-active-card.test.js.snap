// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Main component 1`] = `<div />`;

exports[`Should render MoviePage component 1`] = `<div />`;
